src/main/java/io/codelex/flightplanner/common/CommonInMemoryRepository.java

```diff
@@ -0,0 +1,83 @@
+package io.codelex.flightplanner.common;
+
+import io.codelex.flightplanner.domain.*;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Repository;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.*;
+
+@Repository
+@ConditionalOnProperty(prefix = "flight-planner", name = "store-type", havingValue = "in-memory")
+public class CommonInMemoryRepository implements CommonRepository {
+
+    private final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
+    private final List<Flight> flightList = new ArrayList<>();
+
+    public CommonInMemoryRepository() {
+    }
+
+    public Flight addFlight(AddFlightRequest flightRequest) {
+        int flightId = flightList.size();
+        Airport flightFrom = flightRequest.getFrom();
+        Airport flightTo = flightRequest.getTo();
+        String flightCarrier = flightRequest.getCarrier();
+        LocalDateTime flightDepartureTime = LocalDateTime.parse(flightRequest.getDepartureTime(), formatter);
+        LocalDateTime flightArrivalTime = LocalDateTime.parse(flightRequest.getArrivalTime(), formatter);
+        if (flightDepartureTime.compareTo(flightArrivalTime) >= 0) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Entered times not possible for a correct flight");
+        }
+        Flight flightToAdd = new Flight(flightId, flightFrom, flightTo, flightCarrier, flightDepartureTime, flightArrivalTime);
+        if (flightList.stream().anyMatch(flight -> flight.equals(flightToAdd))) {
```

Comment:
Ar šo tā āķīgi - reāli, ja kāds pārģenerētu equals metodi, tad šis vairs nestrādātu, jo šorīd equals metode ignorē ID. Vari atstāt, bet dzīvē visticamāk notiktu šādi - kāds pievienotu vēl kādu lauku pie Flight, pārģenerētu equals metodi un pat nepamanītu, ka iepriekš viņā nebija ID un šī pārbaude pārstātu strādāt. :D Pareizāk būtu salīdzināt atsevišķā metodē tos laukus, iespējams iekš Flight klases vari uztaisīt metodi isEqualToFlight(Flight anotherFlight) un tur salīdzināt laukus.

Reply:
Aizstaju equals ar areFlightsEqual(Flight flightToCompare) metodi un equals() neizmantoju vispaar, jo vismaz pashreizeejaa variantaa nav nepiecieshama taada metode